co-simulation/fed/mosaic-carma-cloud/src/test/java/org/eclipse/mosaic/fed/carmacloud/ambassador/CarmaCloudMessageAmbassadorTest.java

```diff
@@ -0,0 +1,133 @@
+
+
+/**
+ * Tests for {@link CarmaCloudMessageAmbassador}.
+ */
+public class CarmaCloudMessageAmbassadorTest {
+
+    private final TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    private final TestFileRule testFileRule = new TestFileRule(temporaryFolder).basedir("carmacloud");
+
+    @Rule
+    public RuleChain chain = RuleChain.outerRule(temporaryFolder).around(testFileRule);
+
+    /**
+     * {@link RtiAmbassador} mock.
+     */
+    private RtiAmbassador rtiMock;
+
+    private CarmaCloudMessageAmbassador ambassador;
+    /**
+     * {@link CarmaCloudInstanceManager} mock.
+     */
+    private CarmaCloudInstanceManager instanceManagerMock;
+    /**
+     * {@link CarmaCloudRegistrationReceiver} mock.
+     */
+    private CarmaCloudRegistrationReceiver receiverMock;
+
+    private ArrayList<CarmaCloudRegistrationMessage> registrationMessages;
+
+    @Before
+    public void setUp() throws IOException, NoSuchFieldException, InternalFederateException, IllegalValueException {
+        // Initialize Mocks
+        rtiMock = mock(RtiAmbassador.class);
+        FederateDescriptor handleMock = mock(FederateDescriptor.class);
+        instanceManagerMock = mock(CarmaCloudInstanceManager.class);
+        receiverMock = mock(CarmaCloudRegistrationReceiver.class);
+        File workingDir = temporaryFolder.getRoot();
+        CLocalHost testHostConfig = new CLocalHost();
+        testHostConfig.workingDirectory = workingDir.getAbsolutePath();
+
+        // Mock methods     
+        when(handleMock.getHost()).thenReturn(testHostConfig);
+        when(handleMock.getId()).thenReturn("carmacloud");
+        CarmaCloudRegistrationMessage message = new CarmaCloudRegistrationMessage("carmacloud-id", "carmacloud-url");
+        registrationMessages = new ArrayList<>();
+        registrationMessages.add(message);
+        when(receiverMock.getReceivedMessages()).thenReturn(registrationMessages);
+
+        // Set mocks as ambassador members through reflection or setters
+        ambassador = new CarmaCloudMessageAmbassador(new AmbassadorParameter("carmacloud",
+            temporaryFolder.newFile("carmacloud/carma-cloud_config.json")));
+
+        ambassador.setRtiAmbassador(rtiMock);
+        ambassador.setFederateDescriptor(handleMock);
+        FieldSetter.setField(ambassador, ambassador.getClass().getDeclaredField("carmaCloudRegistrationReceiver"), receiverMock);
+        FieldSetter.setField(ambassador, ambassador.getClass().getDeclaredField("carmaCloudInstanceManager"), instanceManagerMock);
+    }
+
+    @Test
+    public void testInitialize() throws InternalFederateException, IllegalValueException {
+        // Test initialize method
+        ambassador.initialize(0, 100 * TIME.SECOND);
+        verify(rtiMock, times(1)).requestAdvanceTime(0L, 0L, ((byte) 1));
+        // cleanup threads and sockets
+        ambassador.close();
+    }
+
+//    @Test
+//    public void testProcessInteraction() throws InternalFederateException {
```

Comment:
If this test is not being used, lets remove it. If it is being used lets fix and uncomment it